bin/benchmark/Sources/benchmark/Diff/DiffResultsTable.swift

```diff
@@ -0,0 +1,104 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+
+struct DiffResultsTable {
+    static var columns: [(name: String, width: Int)] = [
+        ("Metric", 40),
+        ("Change", 15),
+        ("Before", 20),
+        ("After", 20),
+    ]
+    static var totalWidth: Int {
+        return columns.reduce(0, { $0 + $1.width + 3 }) - 1
+    }
+    
+    private(set) var output: String
+    init(results: DiffResults) {
+        var output = ""
+        
+        let allWarnings = results.analysis.flatMap { $0.warnings ?? [] }
+        for warning in allWarnings {
+            output += "\(warning)\n"
+        }
+        
+        output += "┌\(String(repeating: "─", count: Self.totalWidth))┐\n"
+        output += Self.formattedRow(columnValues: Self.columns.map { $0.name })
+        output += "├\(String(repeating: "─", count: Self.totalWidth))┤\n"
+        
+        var footnoteCounter = 0
+        
+        for analysis in results.analysis {
+            var change: String
+            switch analysis.change {
+                case .same:
+                    change = "no change"
+                case .differentChecksum:
+                    change = "change"
+                case .differentNumeric(let percentage):
+                    change = percentageFormatter.string(from: NSNumber(value: percentage))!
+                case .notApplicable:
+                    change = "n/a"
+            }
+            if let footnotes = analysis.footnotes, !footnotes.isEmpty {
+                let footNoteSuffix = (footnoteCounter ..< footnoteCounter+footnotes.count).map { Self.superscriptCharacters[$0] }.joined(separator: ",")
```

Comment:
Nit: the superscripts look a bit odd with the non-superscript comma separators. I'm wondering if instead of superscripts altogether, we could present the detailed information below the table by repeating the metric titles.